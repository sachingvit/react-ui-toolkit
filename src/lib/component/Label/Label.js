import React from 'react'
import NoContent from '../NoContent/NoContent'
// const { useState, useEffect } = React
function Label(props) {
    const { layoutClasses, label, labelClasses, id } = props

    if (label || label === "") {
        return <NoContent />
    }

    return (
        <div className={layoutClasses}>
            <label className={labelClasses} for={id} id={`label-${id}`}>{label}</label>
        </div>
    )
}

//export Label;
export default Label
