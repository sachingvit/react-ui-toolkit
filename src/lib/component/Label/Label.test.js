import Label from './Label'
import TestRenderer from 'react-test-renderer';

describe('Component > Label', () => {

    let props;

    beforeEach(() => {
        props = {
            label: 'test label string',
            id: '1'
        }
    })

    it('render componet', () => {
        const componentProps = {
            ...props
        }
        const component = <Label {...componentProps} />
        expect(TestRenderer.create(component).toJSON()).toMatchSnapshot()
    })

    it('render without label', () => {
        const component = <Label />
        expect(TestRenderer.create(component).toJSON()).toMatchSnapshot()
    })
})