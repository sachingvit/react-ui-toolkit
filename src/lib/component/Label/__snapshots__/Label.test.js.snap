// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component > Label render componet 1`] = `null`;

exports[`Component > Label render without label 1`] = `
<div>
  <label
    id="label-undefined"
  />
</div>
`;
